src/snapshot.rs

```diff
@@ -133,15 +95,11 @@ impl<P: KeyTrait, V: Clone> Snapshot<P, V> {
     }
 
     pub fn iter(&self) -> Result<Iter<P, V>, TrieError> {
-        // Check if the snapshot is already closed
-        self.is_closed()?;
         Ok(Iter::new(self.root.as_ref()))
```

Comment:
WDYT of getting rid of these Result too in this PR?

Reply:
Good point! Will do 👍

Reply:
Let me handle this in a follow-up removing `Result`'s altogether.